h2/src/main/org/h2/mvstore/cache/CacheLongKeyLIRS.java

```diff
@@ -602,6 +604,21 @@ public void trimNonResidentQueue() {
          */
         private int stackMoveCounter;
 
+        /*
+         * Holds entries that were concurrently get()
+         */
+        private final ConcurrentSkipListSet<Entry<V>> concAccess;
+
+        /*
+         * Serialize access to this segments
+         */
+        private final ReentrantLock l;
+
+        /*
+         * Used as null value for ConcurrentSkipListSet
+         */
+        private final Entry<V> ENTRY_NULL = new Entry<>();
```

Comment:
I think it would be better to convert `misses` to `AtomicLong` and avoid a lock for a `null` value completely. In that case this dummy value and related tricks will not be required.

Reply:
Make sense. Will do.

Reply:
> and avoid a lock for a null value completely
Actually there is no separate lock for this null value - most of the time processing is done inside lock that is currently active. But i agree that it looks a bit 'artificial'.